### PR TITLE
Ignore tests in Rust code instead of Nix config

### LINKED ISSUE #3475: migrate test exclusions from nix to rust
instead of excluding tests in the nix expressions we can replace them with macro annotations on the relevant tests. this will improve consistency for all the various the devs and CI run tests.

### PR DESCRIPTION
**Pull request by ThetaSinner**
### Summary

Closes #3475

### TODO:
- [ ] CHANGELOGs updated with appropriate info

### CODE REVIEW

File: crates/holochain/tests/test_utils/mod.rs [hunk 1]
```diff
@@ -370,16 +369,15 @@ async fn check_timeout_named<T>(
     name: &'static str,
     response: impl Future<Output = std::io::Result<T>>,
     timeout_millis: u64,
-) -> T {
-    // FIXME(stefan): remove this multiplier once it's faster on self-hosted CI
-    let timeout_millis = timeout_millis * 4;
-    match tokio::time::timeout(std::time::Duration::from_millis(timeout_millis), response).await {
-        Ok(response) => response.unwrap(),
-        Err(e) => {
-            panic!(
```
Comment (ThetaSinner) [inline crates/holochain/tests/test_utils/mod.rs:379]: When there are multiple call paths to this that use `<unnamed>`, it's hard to work out which one. Same as we've done other places, bubble the error up so you can see where the test got to

File: nix/modules/holochain.nix [hunk 2]
```diff
@@ -74,26 +74,6 @@
       });
 
       holochainTestsNextestArgs =
-        let
-          # e.g.
-          # "conductor::cell::gossip_test::gossip_test"
-          disabledTests = [
-            "core::ribosome::host_fn::remote_signal::tests::remote_signal_test"
```
Comment (ThetaSinner) [inline nix/modules/holochain.nix:81]: Doesn't seem flaky, trying it out

File: nix/modules/holochain.nix [hunk 3]
```diff
@@ -74,26 +74,6 @@
...
-            "new_lair::test_new_lair_conductor_integration"
```
Comment (ThetaSinner) [inline nix/modules/holochain.nix:82]: Wasn't running for me, it runs now and doesn't seem to be flaky, trying it out

File: nix/modules/holochain.nix [hunk 4]
```diff
@@ -74,26 +74,6 @@
...
-            "conductor::cell::gossip_test::gossip_test"
```
Comment (ThetaSinner) [inline nix/modules/holochain.nix:83]: Actually is flaky, marked as ignored

File: nix/modules/holochain.nix [hunk 5]
```diff
@@ -74,26 +74,6 @@
...
-          ] ++ (lib.optionals (pkgs.system == "x86_64-darwin") [
-            "test_reconnect"
```
Comment (ThetaSinner) [inline nix/modules/holochain.nix:85]: Not flaky to me, trying it out

File: nix/modules/holochain.nix [hunk 6]
```diff
@@ -74,26 +74,6 @@
...
-            "timeout::tests::kitsune_backoff"
```
Comment (ThetaSinner) [inline nix/modules/holochain.nix:86]: Not flaky to me, trying it out

File: nix/modules/holochain.nix [hunk 7]
```diff
@@ -74,26 +74,6 @@
...
-            "test_util::switchboard::tests::transitive_peer_gossip"
```
Comment (ThetaSinner) [inline nix/modules/holochain.nix:87]: Not even building this any more, just drop reference to it

File: nix/modules/holochain.nix (hunk 5 above)
Comment (ThetaSinner) [reply nix/modules/holochain.nix:85]: This isn't flaky on macos, it's actually broken there

File: nix/modules/holochain.nix (hunk 3 above)
Comment (ThetaSinner) [reply nix/modules/holochain.nix:82]: Not flaky on macos, actually broken

File: crates/holochain/tests/test_utils/mod.rs [hunk 8]
```diff
@@ -370,34 +374,34 @@ async fn check_timeout_named<T>(
...
-                "{}: Timed out on request after {}: {}",
-                name, timeout_millis, e
-            );
-        }
+) -> std::io::Result<T> {
+    let timeout_millis = timeout_millis;
```
Comment (jost-s) [inline crates/holochain/tests/test_utils/mod.rs:378]: assignment not needed any more
```suggestion
```

**Review by jost-s — APPROVED** [review]
Yeah!

**Comment by jost-s** [comment]
Ah oh, this is still draft. Okay, consider it a draft review!

**Comment by ThetaSinner** [comment]
Sorry yeah, I thought I was closer because the tests were working locally :)

**Review by maackle — APPROVED** [review]
:clap:

**Pull request merged**